### PR TITLE
Feat: add possible_dates() method to Trip

### PR DESCRIPTION
**Pull request by GregoryJFischer**
### What'd ya do to the code?
Added a new method that returns an array of possible dates that work for any user that is on the trip

### Select One Thing
- [x] This broke nothing

### Type of change
- [x] New feature (non-breaking change which adds functionality)

### Checklist:
- [x] If this code needs to be tested, all tests are passing
- [x] The code runs locally
- [x] I reviewed my code before pushing

### What's next?
We could try breaking this monster up into helper methods, but it works atm, so I think it's fine for MVP, maybe also try to be more robust in edge case / sad path with the test

### CODE REVIEW

**Comment by omegaeye-he** [comment]
Great job!!! This PR is perfect.

File: api/tests/test_models.py [hunk 1]
```diff
@@ -39,3 +39,41 @@ def test_trip_attributes(self):
...
+                'end_date': 10
+            },
+            {   'start_date': 20,
+                'end_date': 29
+            }
+        ])
+
+        TripUser.objects.create(user_id=user3.id, trip_id=trip.id, start_date=100, end_date=101)
+
+        self.assertEqual(trip.possible_dates(), [])
```
Comment (samueldevine) [inline api/tests/test_models.py:79]: nice job with these tests; looks like they cover all of the scenarios we talked through

File: api/models.py [hunk 2]
```diff
@@ -17,6 +17,55 @@ class Trip(models.Model):
...
+            all_users = list(all_users)
+        else:
+            all_users = [all_users]
+
+        other_users = []
+        for user in all_users:
+            if user not in other_users:
+                other_users.append(user)
+
+        other_users.pop(0)
```
Comment (samueldevine) [inline api/models.py:41]: i'm not sure how yet, but I definitely think this section is a great target for a refactor later on, either moving it to a helper method or just shortening it if we can

**Review by samueldevine — APPROVED** [review]
